### PR TITLE
RFE: ghak7 add test for syscall init_module (v2)

### PR DESCRIPTION
**Pull request by rgbriggs**
Hi, sorry about the extra pull request on this feature test.  I had intended to do a supplementary patch on top of the existing ghak7 test to add the ghak37 test, but my local tree had been rebased onto a fresher upstream and a couple of very minor changes had been made.  Close the other pull request if you like.  I don't see how to delete branches in github.

### CODE REVIEW

File: tests/syscall_module/test [hunk 1]
```diff
@@ -37,35 +37,52 @@ system("auditctl -D >& /dev/null");
...
+ok($found_name); # Was the load module found?
+
+$result = system("rmmod $name");
+ok($result, 0); # Did the rmmod succeed?
+system("sleep 1;sync");
+seek($fh_out, 0, 0);
+seek($fh_err, 0, 0);
+$found_name = 0;
+$result = system("ausearch -k $key-unload > $stdout 2> $stderr");
+ok($result, 0); # Did the unlaod search find records?
```
Comment (pcmoore) [inline tests/syscall_module/test]: "unlaod"? ;)

Don't worry about it, this looks fine to me otherwise, I can fix this while merging.

**Comment by pcmoore** [comment]
I'm going hold this until we get the module delete support in the kernel.

**Comment by The-Mule** [comment]
AFAIK this can be merged now, right? 

```
# SUBDIRS="syscall_module" make -e test
make -C tests test
make[1]: Entering directory `/tmp/audit-testsuite/tests'
make[2]: Entering directory `/tmp/audit-testsuite/tests/syscall_module'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/tmp/audit-testsuite/tests/syscall_module'
chmod +x */test
Running as   user    root
        with context unconfined_u:unconfined_r:unconfined_t
        on   system  RHEL7

syscall_module/test .. ok   
All tests successful.
Files=1, Tests=6,  2 wallclock secs ( 0.01 usr  0.00 sys +  0.03 cusr  0.02 csys =  0.06 CPU)
Result: PASS
make[1]: Leaving directory `/tmp/audit-testsuite/tests'

# git branch 
* ghak7-audit-init_module-record-module-name.v2
  master

# ausearch -ts recent
...
time->Fri May  5 08:00:02 2017
type=PROCTITLE msg=audit(1493985602.730:297): proctitle=6D6F6470726F6265006172705F7461626C6573
type=KERN_MODULE msg=audit(1493985602.730:297): name="arp_tables"
type=SYSCALL msg=audit(1493985602.730:297): arch=c000003e syscall=175 success=yes exit=0 a0=fac0c0 a1=85a1 a2=41a2d8 a3=f9d300 items=0 ppid=1609 pid=1610 auid=0 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts0 ses=4 comm="modprobe" exe="/usr/bin/kmod" subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 key="testsuite-1493985602-BbcruVqe-load"
----
time->Fri May  5 08:00:03 2017
type=PROCTITLE msg=audit(1493985603.764:298): proctitle=726D6D6F64006172705F7461626C6573
type=KERN_MODULE msg=audit(1493985603.764:298): name="arp_tables"
type=SYSCALL msg=audit(1493985603.764:298): arch=c000003e syscall=176 success=yes exit=0 a0=897328 a1=800 a2=7f015000eac0 a3=7fffaf591f70 items=0 ppid=1600 pid=1617 auid=0 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=pts0 ses=4 comm="rmmod" exe="/usr/bin/kmod" subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 key="testsuite-1493985602-BbcruVqe-unload"
...
```

**Comment by pcmoore** [comment]
Merged via 7ee84e46f74b4d68038b498ed0f846f05b65fa40 and c909c9d7b97580d31553f2ee5ec3200b502d00d7.

**Pull request closed without merging**